src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,272 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindDateCommand;
+import duke.command.FindDescriptionCommand;
+import duke.command.ListCommand;
+import duke.command.TaskCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyInputException;
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidInputException;
+import duke.exception.InvalidTaskException;
+import duke.exception.UnknownInputException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskType;
+import duke.task.Todo;
+
+/**
+ * The Parser deals with making sense of the user command.
+ */
+public class Parser {
+    /**
+     * Returns the command after parsing the input.
+     *
+     * @param input the input.
+     * @return Command the command.
+     * @throws DukeException If the input cannot be parsed.
+     */
+    public static Command parse(String input) throws DukeException {
+        String keyword = input.split(" ")[0];
+        switch (keyword) {
+        case "bye":
+            return new ByeCommand();
+        case "list":
+            return new ListCommand();
+        case "done":
+            return getDoneCommand(input);
+        case "delete":
+            return getDeleteCommand(input);
+        case "todo":
+            return getTodoTaskCommand(input);
+        case "event":
+            return getEventTaskCommand(input);
+        case "deadline":
+            return getDeadlineTaskCommand(input);
+        case "date":
+            return getFindDateCommand(input);
+        case "find":
+            return getFindCommand(input);
+        default:
+            throw new UnknownInputException();
+        }
+    }
+
+    /**
+     * Returns done command.
+     *
+     * @param input the input.
+     * @return the done command.
+     * @throws EmptyInputException   If the task number is not in input.
+     * @throws UnknownInputException If the task number cannot be parsed.
+     */
+    private static DoneCommand getDoneCommand(String input) throws EmptyInputException,
```

Comment:
Maybe this can be placed inside DoneCommand?

Reply:
Sorry but don't know what 'this' refers to